src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -72,32 +73,72 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
         }
 
         var selectBuilder = GetSelectBuilderForDocumentTable(tablePrefix, providerName);
-
         try
         {
             var selectCommand = connection.CreateCommand();
             selectCommand.CommandText = selectBuilder.ToSqlString();
 
             using var result = await selectCommand.ExecuteReaderAsync();
-
-            // at this point the query succeeded and the table exists
-            return DbConnectionValidatorResult.DocumentFound;
+            if (!isDefaultShell)
+            {
+                // The 'Document' table exists.
+                return DbConnectionValidatorResult.DocumentTableFound;
+            }
+
+            var columns = Enumerable.Range(0, result.FieldCount).Select(result.GetName);
+            if (!columns.Any(c => c == "Type") || !columns.Any(c => c == "Content") || !columns.Any(c => c == "Version"))
+            {
+                // The 'Document' table exists with another schema.
+                return DbConnectionValidatorResult.DocumentTableFound;
+            }
         }
         catch
         {
-            // at this point we know that the document table does not exist
+            // The 'Document' table does not exist.
+            return DbConnectionValidatorResult.DocumentTableNotFound;
+        }
+
+        selectBuilder = GetSelectBuilderForShellDescriptorDocument(tablePrefix, providerName);
+        try
+        {
+            var selectCommand = connection.CreateCommand();
+            selectCommand.CommandText = selectBuilder.ToSqlString();
 
-            return DbConnectionValidatorResult.DocumentNotFound;
+            using var result = await selectCommand.ExecuteReaderAsync();
+            if (!result.HasRows)
+            {
+                // The 'Document' table exists with no 'ShellDescriptor' document.
+                return DbConnectionValidatorResult.ShellDescriptorDocumentNotFound;
```

Comment:
It may be better to return ShellDescriptorFound here and flip the above if statement. I think Found is a bit easier to follow.

Reply:
I will look at it

Reply:
Okay I'm not against but then we would need to also check for `ShellDescriptorDocumentFound` in both tenant validator and setup service, to not validate in that case.

Then if the fipped if is false what we will return that will allow the validation to pass, not `DocumentTableNotFound` as you will say it is wrong and you will be right, so we would need to still keep `ShellDescriptorDocumentNotFound`, as you want.

Let me know what you prefer.

Reply:
I think you don't need to change anything if you code look like this

```
if (result.HasRows)
{
    // The 'Document' table exists with no 'ShellDescriptor' document.
    return DbConnectionValidatorResult.ShellDescriptorDocumentFound;
}
```
So you don't need to change other places

Reply:
If I don't change anything else, if the the ShellDescriptor is not found we would return below DocumentTableFound which will not let pass the validation.

Reply:
Alight.  I would leave it as is then.